ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/LedgerApiTestTool.scala

```diff
@@ -174,22 +176,60 @@ object LedgerApiTestTool {
     }
   }
 
+  private[this] def newSequentialLedgerCasesRunner(
+      config: Config,
+      cases: Vector[LedgerTestCase],
+  )(implicit executionContext: ExecutionContext): Future[LedgerTestCasesRunner] =
+    createLedgerCasesRunner(config, cases, concurrentTestRuns = 1)
+
   private[this] def newLedgerCasesRunner(
       config: Config,
-      cases: Iterable[LedgerTestCase],
-      concurrencyOverride: Option[Int] = None,
-  ): LedgerTestCasesRunner =
-    new LedgerTestCasesRunner(
-      new LedgerSessionConfiguration(
-        config.participants,
-        config.tlsConfig,
-        config.partyAllocation,
-        config.shuffleParticipants,
-      ),
-      cases.toVector,
-      identifierSuffix,
-      config.timeoutScaleFactor,
-      concurrencyOverride.getOrElse(config.concurrentTestRuns),
-      uploadDars = config.uploadDars,
+      cases: Vector[LedgerTestCase],
+  )(implicit executionContext: ExecutionContext): Future[LedgerTestCasesRunner] =
+    createLedgerCasesRunner(config, cases, config.concurrentTestRuns)
+
+  private[this] def createLedgerCasesRunner(
+      config: Config,
+      cases: Vector[LedgerTestCase],
+      concurrentTestRuns: Int,
+  )(implicit executionContext: ExecutionContext): Future[LedgerTestCasesRunner] = {
+    initializeParticipantChannels(config.participants, config.tlsConfig).asFuture.map(
+      participants =>
+        new LedgerTestCasesRunner(
+          testCases = cases,
+          participants = participants,
+          partyAllocation = config.partyAllocation,
+          shuffleParticipants = config.shuffleParticipants,
+          timeoutScaleFactor = config.timeoutScaleFactor,
+          concurrentTestRuns = concurrentTestRuns,
+          uploadDars = config.uploadDars,
+          identifierSuffix = identifierSuffix,
+        )
     )
+  }
+
+  private def initializeParticipantChannels(
+      participants: Vector[(String, Int)],
+      tlsConfig: Option[TlsConfiguration],
+  )(implicit executionContext: ExecutionContext): Resource[ExecutionContext, Vector[Channel]] = {
```

Comment:
maybe worth factoring out the code for a single participant? it seems a bit odd to only have the version that accepts a `Vector` if it treats all of them independently.

Reply:
Why not.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/8607/commits/3093123330cfb7a6db61532d43ebeed33bb6c144